File: src/main/java/com/youdeyiwu/enums/forum/PostReviewStateEnum.java
package com.youdeyiwu.enums.forum;

import lombok.Getter;

/**
 * post review state.
 * If you need to add an enumeration, please append it at the end.
 * This is because the enumeration is stored in natural order, not by name.
 *
 * @author dafengzhen
 */
@Getter
public enum PostReviewStateEnum {

  /**
   * approved (Default).
   */
  APPROVED,

  /**
   * rejected (The forum administrators, section administrators, and post authors have access).
   */
  REJECTED,

  /**
   * pendingReview (The forum administrators, section administrators, and post authors have access).
   */
  PENDING_REVIEW,

}

File: app/lib/enums/post_review_state_enum.dart
import 'package:json_annotation/json_annotation.dart';

enum PostReviewStateEnum {
  /// approved (Default).
  @JsonValue('APPROVED')
  approved,

  /// rejected (The forum administrators, section administrators, and post authors have access).
  @JsonValue('REJECTED')
  rejected,

  /// pending review (The forum administrators, section administrators, and post authors have access).
  @JsonValue('PENDING_REVIEW')
  pendingReview,
}
